conans/client/rest/conan_requester.py

```diff
@@ -58,14 +60,36 @@ def _add_kwargs(self, url, kwargs):
         return kwargs
 
     def get(self, url, **kwargs):
-        return self._requester.get(url, **self._add_kwargs(url, kwargs))
+        return self._call_method("get", url, **kwargs)
 
     def put(self, url, **kwargs):
-        return self._requester.put(url, **self._add_kwargs(url, kwargs))
+        return self._call_method("put", url, **kwargs)
 
     def delete(self, url, **kwargs):
-        return self._requester.delete(url, **self._add_kwargs(url, kwargs))
+        return self._call_method("delete", url, **kwargs)
 
     def post(self, url, **kwargs):
-        return self._requester.post(url, **self._add_kwargs(url, kwargs))
+        return self._call_method("post", url, **kwargs)
 
+    def _call_method(self, method, url, **kwargs):
+        env = self._get_env()
+        with tools.environment_append(env):
```

Comment:
It is not clear how this clears the environment, as it only appends or replace

Reply:
You are right, how the hell is working the test?

Reply:
I see, os.environ is not a pure dict. So it is modifying the real env too. I'll change this.